### PR TITLE
Add fixes for osx-arm64 migration

### PR DESCRIPTION
**Pull request by erykoff**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by erykoff** [comment]
cc/ @isuruf

File: recipe/build.sh [hunk 1]
```diff
@@ -54,6 +54,16 @@ if [[ "$target_platform" == "linux-aarch64" || "$target_platform" == "osx-arm64"
   TEST_CMD=""
 fi
 
+if [[ "$target_platform" == "osx-arm64" ]]; then
+  ARCH_OPTS_SINGLE="--enable-neon"
+  ARCH_OPTS_DOUBLE="--enable-neon"
+  ARCH_OPTS_LONG_DOUBLE="--enable-long-double"
```
Comment (erykoff) [inline recipe/build.sh:60]: osx-arm64 supports `long double` but it's the same 64-bit width as `double`.  Does that matter?

Comment (isuruf) [reply recipe/build.sh:60]: I guess so. Let's disable that

**Comment by erykoff** [comment]
It seems that `openmpi` is failing on arm64.  I'll look into that.  Otherwise, I confirmed that `osx_arm64_mpimpich` works and passes all tests.

**Comment by isuruf** [comment]
For openmpi, you need https://github.com/conda-forge/mpi4py-feedstock/blob/master/recipe/meta.yaml#L16

File: recipe/build.sh [hunk 2]
```diff
@@ -57,6 +58,16 @@ if [[ "$target_platform" == "linux-aarch64" ]]; then
   TEST_CMD=""
 fi
 
+if [[ "$target_platform" == "osx-arm64" ]]; then
+  ARCH_OPTS_SINGLE="--enable-neon"
+  ARCH_OPTS_DOUBLE="--enable-neon"
+  ARCH_OPTS_LONG_DOUBLE=""
```
Comment (isuruf) [inline recipe/build.sh:64]: This wouldn't disable the long double library, this would override the double library which we do not want

**Comment by erykoff** [comment]
@isuruf is this ready?

**Comment by isuruf** [comment]
Yes, @conda-forge/fftw

File: recipe/build.sh [hunk 3]
```diff
@@ -33,27 +33,41 @@ TEST_CMD="eval cd tests && make check-local && cd -"
 if [[ "$target_platform" == "linux-64" ]] || [[ "$target_platform" == "linux-32" ]] || [[ "$target_platform" == "osx-64" ]]; then
   ARCH_OPTS_SINGLE="--enable-sse --enable-sse2 --enable-avx"
```
Comment (beckermr) [inline recipe/build.sh:34]: IDK when these flags snuck in, but I didn't think avx was allowed given our generic builds.

**Review by beckermr — APPROVED** [review]
One comment on us using AVX which I didn't think we could but clearly ppl have been ok. LGTM!

File: recipe/build.sh (hunk 3 above)
Comment (isuruf) [reply recipe/build.sh:34]: They have runtime cpu detection. This enables building with sse2

Comment (beckermr) [reply recipe/build.sh:34]: Ahh great!

**Pull request merged**